articles/142_idl.md

```diff
@@ -35,3 +35,93 @@ Further it describes how the interfaces are used to generate code in C, C++ and
 <div class="alert alert-warning" markdown="1">
   <b>TODO:</b> All of this too
 </div>
+
+### Constraint Checking
+
+Generated code for messages is only guaranteed to enforce constraints when a message is published.
+For example, there might not be an error when a field constrained to be a fixed size array is assigned an array with the wrong size.
+This could be inconsistent across client library implementations depending on language features and performance cost.
+In the future any client library may add additional constraint checking when a field is set or modified.
+Users of generated message code should assume constraints could be enforced at any time to be compatible with such a change.
+
+### Type Mapping
+
+The next table defines how IDL types are mapped to the following programming languages:
+
+* C11 (or higher)
+* C++11 (or higher)
+* Python 3 (or higher)
+
+| IDL type    | C type           | C++ type                       | Python type          |
+| --------    | -----------      | ------------------------------ | -------------------- |
+| float       | float            | float                          | float                |
+| double      | double           | double                         | float                |
+| long double | long double      | long double                    | float                |
```

Comment:
Thoughts on not initially supporting `long double`? It seems inconsistent across platforms and DDS vendors.

Visual Studio `typedef`'d `long double` to `double` which is 64bit
https://docs.microsoft.com/en-us/cpp/c-language/type-long-double?view=vs-2017

Connext works around this by mapping `DDS_LongDouble` to a `char` array (of size 16?) on some platforms and `long double` on others.
https://community.rti.com/static/documentation/connext-dds/5.2.3/doc/manuals/connext_dds/html_files/RTI_ConnextDDS_CoreLibraries_UsersManual/Content/UsersManual/Translations_for_IDL_Types.htm#Note2_LongDouble

Opensplice doesn't appear to support it. The grammar on [page 589 / pdf page 617](http://download.prismtech.com/docs/Vortex/pdfs/OpenSplice_refman_CPP.pdf)
 only defines `<floating_pt_type>` as `"float"` and `"double"`.

Fast-CDR uses the C++ type `long double` and serializes it as 16 bytes on the wire. https://github.com/eProsima/Fast-CDR/blob/1df42bbd1ea86a9c99bb8fa72104f19872235986/src/cpp/Cdr.cpp#L528

Reply:
My understanding from original discussion was that we *should* support all IDL types. @dirk-thomas is that correct?

Looking a bit more into it:
- It looks like Connext and Fast-RTPS agree that it should be a 16-byte thing on the wire

The issues we face right now:
- FastRTPS doesn't have special handling to account for the fact that the type has a different size on windows
- OpenSplice doesn't support this type

Looking at Opensplice it appears that wchar and wstring are not supported either https://github.com/ADLINK-IST/opensplice/blob/36e4f9c250a640ade72d5af92084f5f5243ce96c/src/tools/idlpp/code/idl_unsupported.h.

Not convinced it should be a blocker for supporting it though. We may just need to have implementation specific limitations or workarounds

Reply:
after talking with @mikaelarguedas I think this can be resolved by saying `long double `will be supported, but users will have to be cautious of middleware support for it.

I'm not sure if we should use the C and C++ type `long double` on all platforms, but that seems like the easiest thing to do for now.

Reply:
Added warning about `long double` on some platforms and middleware in 
2082ba0